x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/datafeed/extractor/chunked/ChunkedDataExtractor.java

```diff
@@ -182,19 +170,97 @@ public void cancel() {
         isCancelled = true;
     }
 
-    private class DataSummary {
+    ChunkedDataExtractorContext getContext() {
+        return context;
+    }
+
+    private class DataSummaryFactory {
+
+        /**
+         * If there are aggregations, an AggregatedDataSummary object is created. It returns a ScrollingDataSummary otherwise.
+         *
+         * By default a DatafeedConfig with aggregations, should already have a manual ChunkingConfig created.
+         * However, the end user could have specifically set the ChunkingConfig to AUTO, which would not really work for aggregations.
+         * So, if we need to gather an appropriate chunked time for aggregations, we can utilize the AggregatedDataSummary
+         *
+         * @return DataSummary object
+         * @throws IOException when timefield range search fails
+         */
+        private DataSummary buildDataSummary() throws IOException {
+            if (context.hasAggregations) {
```

Comment:
How about extracting both branches into their own methods here? i.e. `createDataSummaryForAggregations` and `createDataSummaryForScroll`.

Reply:
For sure.